### PR TITLE
Update mimalloc to 2.0.6 and disable the overrides

### PR DESCRIPTION
**Pull request by gbaraldi**
This is for experimenting with different allocators in julia.

### CODE REVIEW

File: M/mimalloc/build_tarballs.jl [hunk 1]
```diff
@@ -3,12 +3,12 @@
...
 
 name = "mimalloc"
-version = v"2.0.1"
+version = v"2.0.6"
 
 # Collection of sources required to complete build
 sources = [
     GitSource("https://github.com/microsoft/mimalloc.git",
-              "8e35ccc43be293a9bfd6e63da310a79c235d25d9"),
+              "82712f4a8f038a7fb4df2790f4c3b7e3ed9e219b"),
```
Comment (giordano) [inline M/mimalloc/build_tarballs.jl:11]: Where did you get that?  https://github.com/microsoft/mimalloc/commit/82712f4a8f038a7fb4df2790f4c3b7e3ed9e219b

Comment (giordano) [reply M/mimalloc/build_tarballs.jl:11]: I presume you want https://github.com/microsoft/mimalloc/commit/f2712f4a8f038a7fb4df2790f4c3b7e3ed9e219b?

Comment (gbaraldi) [reply M/mimalloc/build_tarballs.jl:11]: That's a copy pasting failure :|

**Comment by giordano** [comment]
If I'm reading https://github.com/microsoft/mimalloc/pull/400 correctly, the patch can be dropped.

File: M/mimalloc/build_tarballs.jl [hunk 2]
```diff
@@ -3,38 +3,29 @@
...
 name = "mimalloc"
-version = v"2.0.1"
+version = v"2.0.6"
 
 # Collection of sources required to complete build
 sources = [
     GitSource("https://github.com/microsoft/mimalloc.git",
-              "8e35ccc43be293a9bfd6e63da310a79c235d25d9"),
+              "f2712f4a8f038a7fb4df2790f4c3b7e3ed9e219b"),
     DirectorySource("./bundled"),
```
Comment (giordano) [inline M/mimalloc/build_tarballs.jl:12]:
```suggestion
```

**Pull request merged**